src/Npgsql/NpgsqlConnection.cs

```diff
@@ -198,21 +194,23 @@ void GetPoolAndSettings()
                 }
 
                 // Maybe pooling is off
-                if (!Settings.Pooling)
-                    return;
-
-                // Connstring may be equivalent to one that has already been seen though (e.g. different
-                // ordering). Have NpgsqlConnectionStringBuilder produce a canonical string representation
-                // and recheck.
-                var canonical = Settings.ConnectionString;
-                if (pools.TryGetValue(canonical, out _pool))
-                    pools[_connectionString] = _pool;
-                else
+                if (Settings.Pooling)
```

Comment:
As above - let's check if pooling is off and return immediately rather than checking whether it's on and further nesting the code.